app/controllers/memberships_controller.rb

```diff
@@ -15,6 +16,17 @@ def create
     end
   end
 
+  def edit
+  end
+
+  def update
+    if @membership.update membership_params
```

Comment:
We're not checking here that the membership params only accept allowed `roles` and `applications`. So in theory, this could be spoofed and users could assign extra permissions.